Src/Couchbase.Linq/BucketDataStore.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Couchbase.Core;
+using Couchbase.N1QL;
+
+namespace Couchbase.Linq
+{
+    /// <summary>
+    /// A set of lower-level methods for other bucket-related functionality
+    /// </summary>
+    internal class BucketDataStore : IDataStore
+    {
+        private readonly IBucket _bucket;
+
+        internal BucketDataStore(IBucket bucket)
+        {
+            _bucket = bucket;
+        }
+
+        /// <summary>
+        /// Execute a N1QL function. Optionally specify parameters
+        /// </summary>
+        public IQueryResult<T> Execute<T>(string statement, params object[] parameters)
+        {
+            var query = new QueryRequest(statement)
+                .AddPositionalParameter(parameters);
+            return _bucket.Query<T>(query);
```

Comment:
On another note, we might want to change the return type to directly return IEnumerable<T> and throw CouchbaseExceptions on errors.  This would be more consistent with the approach in the rest of the library.